homeassistant/components/overkiz/climate_entities/hitachi_air_to_air_heat_pump.py

```diff
@@ -0,0 +1,360 @@
+
+    @property
+    def current_temperature(self) -> int | None:
+        """Return current temperature."""
+        if (
+            state := self.device.states[ROOM_TEMPERATURE_STATE[self.protocol]]
+        ) and state.value_as_int:
+            return state.value_as_int
+
+        return None
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new temperature."""
+        temperature = cast(float, kwargs.get(ATTR_TEMPERATURE))
+        await self._global_control(target_temperature=int(temperature))
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Return the current preset mode, e.g., home, away, temp."""
+        if (
+            self.protocol == Protocol.OVP
+            and (state := self.device.states[OverkizState.CORE_HOLIDAYS_MODE_STATE])
+            and state.value_as_str
+        ):
+            if state.value_as_str == OverkizCommandParam.ON:
+                return PRESET_HOLIDAY_MODE
+
+            if state.value_as_str == OverkizCommandParam.OFF:
+                return PRESET_NONE
+
+        if (
+            self.protocol == Protocol.HLRR_WIFI
+            and (state := self.device.states[LEAVE_HOME_STATE[self.protocol]])
+            and state.value_as_str
+        ):
+            if state.value_as_str == OverkizCommandParam.ON:
+                return PRESET_HOLIDAY_MODE
+
+            if state.value_as_str == OverkizCommandParam.OFF:
+                return PRESET_NONE
+
+        return None
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+        # OVP uses setHolidays, where HLRRWIFI uses an extra parameter in global control
+        if self.protocol == Protocol.HLRR_WIFI:
+            if preset_mode == PRESET_HOLIDAY_MODE:
+                await self._global_control(leave_home=OverkizCommandParam.ON)
+
+            if preset_mode == PRESET_NONE:
+                await self._global_control(leave_home=OverkizCommandParam.OFF)
+
+        if self.protocol == Protocol.OVP:
+            if preset_mode == PRESET_HOLIDAY_MODE:
+                await self.executor.async_execute_command(
+                    OverkizCommand.SET_HOLIDAYS,
+                    OverkizCommandParam.ON,  # add to OverkizCommand
+                )
+            if preset_mode == PRESET_NONE:
+                await self.executor.async_execute_command(
+                    OverkizCommand.SET_HOLIDAYS,
+                    OverkizCommandParam.OFF,  # add to OverkizCommand
+                )
+
+    def _control_backfill(
+        self, value: str | None, state_name: str, fallback_value: str
+    ) -> str:
+        """Overkiz doesn't accept commands with undefined parameters. This function is guaranteed to return a `str` which is the provided `value` if set, or the current device state if set, or the provided `fallback_value` otherwise."""
+        if value:
+            return value
+        state = cast(State, self.device.states[state_name])
```

Comment:
Why do you need to cast it here, if you check for `if state` below?

Reply:
Good catch, there is no need.